package com.capgemini.studentexception.main;

import com.capgemini.studentexception.exception.StudentException;

public class Student {
	private int studentRegNo;
	private String studentName;
	private int studentAge;
	private String studentCourse;

	public Student() {
		super();
	}

	public Student(int studentRegNo, String studentName, int studentAge, String studentCourse) {
		super();
		this.studentRegNo = studentRegNo;
		this.studentName = studentName;
		this.studentAge = studentAge;
		this.studentCourse = studentCourse;
	}

	public String getStudentName() {
		if(studentName.matches("^.*[^a-zA-Z ]{9}.*$") == false){
			throw new StudentException("Name is not validexception");
		}
		else
		return studentName;
	}

	public void setStudentName(String studentName) {
		this.studentName = studentName;
	}

	public int getStudentAge() {
		if (15 <= studentAge  && studentAge <= 21)
			return studentAge;
		else
			throw new StudentException("Age is Not within range exception");
	}

	public void setStudentAge(int studentAge) {
		this.studentAge = studentAge;
		
	}

	
	
}